lib/control/control-main.c

```diff
@@ -26,18 +26,61 @@
 #include "control-server.h"
 #include "control-commands.h"
 
-static ControlServer *control_server;
+#include <iv.h>
+#include <iv_event.h>
+
+typedef struct _ControlServerThread
+{
+  GThread *thread;
+  MainLoop *main_loop;
+  const gchar *control_name;
+  struct iv_event stop_requested;
+} ControlServerThread;
+
+static ControlServerThread control_server_thread;
```

Comment:
Instead of tbe static variable,cant we store this in globalconfig? As I see this is strictly tied to onr mainloop instance.

Hmm now as i think of it, it may not be true though, the control socket is probably arching over multiple globalconfig instances. If this is the case, then it should probably be moved to the main program.

Reply:
ok, I can move it to main